Back-port of the following two patches from wireless-testing...

commit 5a2e156203294d61d7963877286b2763cbc02cf7
Author: Ivo van Doorn <ivdoorn@gmail.com>
Date:   Sat Jan 17 20:44:08 2009 +0100

    rt2x00: Add LED_MODE_ASUS support
    
    When the led mode is asus, the activity led mode must
    be registered otherwise the second LED will not be enabled.
    
    Signed-off-by: Ivo van Doorn <IvDoorn@gmail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

commit a064405c724484f80eefdb6754e8f782144e1f91
Author: Andrew Price <andy@andrewprice.me.uk>
Date:   Sat Jan 10 19:38:05 2009 +0000

    rt2400,rt2500: init led_qual for LED_MODE_DEFAULT
    
    Add a check for LED_MODE_DEFAULT so that we use the link LED for rt2400
    and rt2500 devices.
    
    Signed-off-by: Andrew Price <andy@andrewprice.me.uk>
    Acked-by: Ivo van Doorn <IvDoorn@gmail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff -up linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2400pci.c.orig linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2400pci.c
--- linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2400pci.c.orig	2008-12-24 18:26:37.000000000 -0500
+++ linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2400pci.c	2009-01-21 16:06:08.000000000 -0500
@@ -1378,7 +1378,9 @@ static int rt2400pci_init_eeprom(struct 
 	value = rt2x00_get_field16(eeprom, EEPROM_ANTENNA_LED_MODE);
 
 	rt2400pci_init_led(rt2x00dev, &rt2x00dev->led_radio, LED_TYPE_RADIO);
-	if (value == LED_MODE_TXRX_ACTIVITY)
+	if (value == LED_MODE_TXRX_ACTIVITY ||
+	    value == LED_MODE_DEFAULT ||
+	    value == LED_MODE_ASUS)
 		rt2400pci_init_led(rt2x00dev, &rt2x00dev->led_qual,
 				   LED_TYPE_ACTIVITY);
 #endif /* CONFIG_RT2X00_LIB_LEDS */
diff -up linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500usb.c.orig linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500usb.c
--- linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500usb.c.orig	2008-12-24 18:26:37.000000000 -0500
+++ linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500usb.c	2009-01-21 16:06:08.000000000 -0500
@@ -1477,7 +1477,9 @@ static int rt2500usb_init_eeprom(struct 
 	value = rt2x00_get_field16(eeprom, EEPROM_ANTENNA_LED_MODE);
 
 	rt2500usb_init_led(rt2x00dev, &rt2x00dev->led_radio, LED_TYPE_RADIO);
-	if (value == LED_MODE_TXRX_ACTIVITY)
+	if (value == LED_MODE_TXRX_ACTIVITY ||
+	    value == LED_MODE_DEFAULT ||
+	    value == LED_MODE_ASUS)
 		rt2500usb_init_led(rt2x00dev, &rt2x00dev->led_qual,
 				   LED_TYPE_ACTIVITY);
 #endif /* CONFIG_RT2X00_LIB_LEDS */
diff -up linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500pci.c.orig linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500pci.c
--- linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500pci.c.orig	2008-12-24 18:26:37.000000000 -0500
+++ linux-2.6.28.noarch/drivers/net/wireless/rt2x00/rt2500pci.c	2009-01-21 16:06:08.000000000 -0500
@@ -1537,7 +1537,9 @@ static int rt2500pci_init_eeprom(struct 
 	value = rt2x00_get_field16(eeprom, EEPROM_ANTENNA_LED_MODE);
 
 	rt2500pci_init_led(rt2x00dev, &rt2x00dev->led_radio, LED_TYPE_RADIO);
-	if (value == LED_MODE_TXRX_ACTIVITY)
+	if (value == LED_MODE_TXRX_ACTIVITY ||
+	    value == LED_MODE_DEFAULT ||
+	    value == LED_MODE_ASUS)
 		rt2500pci_init_led(rt2x00dev, &rt2x00dev->led_qual,
 				   LED_TYPE_ACTIVITY);
 #endif /* CONFIG_RT2X00_LIB_LEDS */
